src/GafferScene/EditScopeAlgo.cpp

```diff
@@ -0,0 +1,260 @@
+const InternedString g_scale( "scale" );
+const InternedString g_pivot( "pivot" );
+
+SceneProcessorPtr transformProcessor()
+{
+	SceneProcessorPtr result = new SceneProcessor( "TransformEdits" );
+
+	SpreadsheetPtr spreadsheet = new Spreadsheet;
+	result->addChild( spreadsheet );
+	spreadsheet->selectorPlug()->setValue( "${scene:path}" );
+
+	PathFilterPtr pathFilter = new PathFilter;
+	result->addChild( pathFilter );
+	pathFilter->pathsPlug()->setInput( spreadsheet->activeRowNamesPlug() );
+
+	TransformPtr transform = new Transform;
+	result->addChild( transform );
+	transform->inPlug()->setInput( result->inPlug() );
+	transform->filterPlug()->setInput( pathFilter->outPlug() );
+	transform->enabledPlug()->setInput( result->enabledPlug() );
+	transform->spacePlug()->setValue( Transform::Space::ResetLocal );
+
+	for( auto name : { g_translate, g_rotate, g_scale, g_pivot } )
+	{
+		auto plug = transform->transformPlug()->getChild<ValuePlug>( name );
+		spreadsheet->rowsPlug()->addColumn( plug );
+		plug->setInput( spreadsheet->outPlug()->getChild<Plug>( name ) );
+	}
+
+	PlugAlgo::promoteWithName( spreadsheet->rowsPlug(), "edits" );
+	result->outPlug()->setInput( transform->outPlug() );
+
+	return result;
+}
+
+EditScope::ProcessorRegistration g_transformProcessorRegistration( "TransformEdits", transformProcessor );
+
+} // namespace
+
+void GafferScene::EditScopeAlgo::setTransform( Gaffer::EditScope *scope, const ScenePlug::ScenePath &path, const Imath::M44f &matrix )
+{
+	V3f scale, shear, rotate, translate;
+	extractSHRT( matrix, scale, shear, rotate, translate );
+	auto p = acquireTransformPlugs( scope, path );
+	p->translate->setValue( translate );
+	p->rotate->setValue( IECore::radiansToDegrees( rotate ) );
+	p->scale->setValue( scale );
+}
+
+boost::optional<Imath::M44f> GafferScene::EditScopeAlgo::getTransform( const Gaffer::EditScope *scope, const ScenePlug::ScenePath &path )
+{
+	auto p = acquireTransformPlugs( const_cast<EditScope *>( scope ), path, /* createIfNecessary = */ false );
+	if( !p )
+	{
+		return boost::none;
+	}
+	return p->matrix();
+}
+
+bool GafferScene::EditScopeAlgo::hasTransform( const Gaffer::EditScope *scope, const ScenePlug::ScenePath &path )
+{
+	return (bool)acquireTransformPlugs( const_cast<EditScope *>( scope ), path, /* createIfNecessary = */ false );
+}
+
+void GafferScene::EditScopeAlgo::removeTransform( Gaffer::EditScope *scope, const ScenePlug::ScenePath &path )
+{
+	auto p = acquireTransformPlugs( scope, path, /* createIfNecessary = */ false );
+	if( !p )
+	{
+		return;
+	}
+	auto row = p->translate->ancestor<Spreadsheet::RowPlug>();
+	row->parent()->removeChild( row );
+}
+
+Imath::M44f EditScopeAlgo::TransformPlugs::matrix() const
+{
+	const V3f pivot = this->pivot->getValue();
+	M44f result;
+	result.translate( pivot + translate->getValue() );
+	result.rotate( IECore::degreesToRadians( rotate->getValue() ) );
+	result.scale( scale->getValue() );
+	result.translate( -pivot );
+	return result;
+}
```

Comment:
Yeah, it's one of the downsides of having the separate translate/rotate/scale/pivot columns in the Spreadsheet instead of just a single TransformPlug. I did consider adding a static `TransformPlug::giveMeAMatrixFromThisBunchOfPlugsIHappenToHave()` method, but wasn't convinced it was really much of an improvement.